diff --git a/gfx/layers/opengl/DMABUFTextureHostOGL.cpp b/gfx/layers/opengl/DMABUFTextureHostOGL.cpp
--- a/gfx/layers/opengl/DMABUFTextureHostOGL.cpp
+++ b/gfx/layers/opengl/DMABUFTextureHostOGL.cpp
@@ -3,6 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "DMABUFTextureHostOGL.h"
+#include "mozilla/StaticPrefs_gfx.h"
 #include "mozilla/widget/DMABufSurface.h"
 #include "mozilla/widget/DMABufFormats.h"
 #include "mozilla/webrender/RenderDMABUFTextureHost.h"
@@ -180,11 +181,21 @@ void DMABUFTextureHostOGL::PushDisplayItems(
   }
   bool preferCompositorSurface =
       aFlags.contains(PushDisplayItemFlag::PREFER_COMPOSITOR_SURFACE);
-  bool supportsDirectComposition =
-      widget::GetGlobalDMABufFormats()->SupportsDirectComposition(
-          mSurface->GetFormat());
+  const auto surfaceFormat = mSurface->GetFormat();
+  const bool hdrPrefsEnabled =
+      StaticPrefs::gfx_color_management_hdr_force_enabled() ||
+      StaticPrefs::gfx_color_management_hdr();
+  const bool supportsDirectComposition =
+      widget::GetGlobalDMABufFormats()->SupportsDirectComposition(surfaceFormat);
+  const bool forceExternalCompositing =
+      !supportsDirectComposition && preferCompositorSurface &&
+      (mSurface->IsHDRSurface() || surfaceFormat == gfx::SurfaceFormat::P010 ||
+       surfaceFormat == gfx::SurfaceFormat::P016) &&
+      hdrPrefsEnabled;
+  const bool supportsExternalCompositing =
+      supportsDirectComposition || forceExternalCompositing;
 
-  switch (mSurface->GetFormat()) {
+  switch (surfaceFormat) {
     case gfx::SurfaceFormat::R8G8B8X8:
     case gfx::SurfaceFormat::R8G8B8A8:
     case gfx::SurfaceFormat::B8G8R8A8:
@@ -196,7 +207,7 @@ void DMABUFTextureHostOGL::PushDisplayItems(
      aBuilder.PushImage(aBounds, aClip, true, false, aFilter, aImageKeys[0],
                          !(mFlags & TextureFlags::NON_PREMULTIPLIED),
                          wr::ColorF{1.0f, 1.0f, 1.0f, 1.0f},
-                         preferCompositorSurface, supportsDirectComposition);
+                         preferCompositorSurface, supportsExternalCompositing);
       break;
     }
     case gfx::SurfaceFormat::NV12: {
@@ -209,7 +220,7 @@ void DMABUFTextureHostOGL::PushDisplayItems(
           aBounds, aClip, true, aImageKeys[0], aImageKeys[1],
           wr::ColorDepth::Color8, wr::ToWrYuvColorSpace(GetYUVColorSpace()),
           wr::ToWrColorRange(GetColorRange()), aFilter, preferCompositorSurface,
-          supportsDirectComposition);
+          supportsExternalCompositing);
       break;
     }
     case gfx::SurfaceFormat::YUV420: {
@@ -223,7 +234,7 @@ void DMABUFTextureHostOGL::PushDisplayItems(
           aBounds, aClip, true, aImageKeys[0], aImageKeys[1], aImageKeys[2],
           wr::ColorDepth::Color8, wr::ToWrYuvColorSpace(GetYUVColorSpace()),
           wr::ToWrColorRange(GetColorRange()), aFilter, preferCompositorSurface,
-          supportsDirectComposition);
+          supportsExternalCompositing);
       break;
     }
     case gfx::SurfaceFormat::P010:
@@ -236,7 +247,7 @@ void DMABUFTextureHostOGL::PushDisplayItems(
           aBounds, aClip, true, aImageKeys[0], aImageKeys[1],
           wr::ColorDepth::Color10, wr::ToWrYuvColorSpace(GetYUVColorSpace()),
           wr::ToWrColorRange(GetColorRange()), aFilter, preferCompositorSurface,
-          supportsDirectComposition);
+          supportsExternalCompositing);
       break;
     }
     default: {
