diff --git a/gfx/wr/webrender/src/tile_cache/mod.rs b/gfx/wr/webrender/src/tile_cache/mod.rs
--- a/gfx/wr/webrender/src/tile_cache/mod.rs
+++ b/gfx/wr/webrender/src/tile_cache/mod.rs
@@ -1845,10 +1845,20 @@ impl TileCacheInstance {
         // If this primitive is an external image, and supports being used
         // directly by a native compositor, then lookup the external image id
         // so we can pass that through.
-        let mut external_image_id = if flags.contains(PrimitiveFlags::SUPPORTS_EXTERNAL_COMPOSITOR_SURFACE)
-            && image_rendering == ImageRendering::Auto {
+        let supports_external_compositor_surface =
+            flags.contains(PrimitiveFlags::SUPPORTS_EXTERNAL_COMPOSITOR_SURFACE);
+        let block_external_for_wide_yuv = matches!(
+            &dependency,
+            ExternalSurfaceDependency::Yuv {
+                channel_bit_depth,
+                ..
+            } if *channel_bit_depth > 8
+        );
+        let mut external_image_id = if supports_external_compositor_surface &&
+            image_rendering == ImageRendering::Auto &&
+            !block_external_for_wide_yuv {
             resource_cache.get_image_properties(api_keys[0])
                 .and_then(|properties| properties.external_image)
                 .and_then(|image| Some(image.id))
         } else {
             None
